### PR TITLE
GUI-2544: Implement copy alarm dialog on alarms landing page

### PR DESCRIPTION
**Pull request by kamalgill**
Implements https://eucalyptus.atlassian.net/browse/GUI-2544

Also fixes:
- https://eucalyptus.atlassian.net/browse/GUI-2570
- https://eucalyptus.atlassian.net/browse/GUI-2573

### CODE REVIEW

**Comment by dkavanagh** [comment]
I'll review this when all 3 tickets are in review for me. Otherwise, I'll just have to go back over it again.

**Comment by dkavanagh** [comment]
hal, retest this please

File: eucaconsole/static/js/widgets/create-alarm-modal/create-alarm.js [hunk 1]
```diff
@@ -146,13 +154,43 @@ angular.module('CreateAlarmModal', [
...
+                $scope.invalidDimensions = false;
                 $scope.namespace = alarm.namespace;
                 $scope.resourceType = attrs.resourceType;
                 $scope.resourceId = attrs.resourceId;
+                if (attrs.editDimensions && attrs.alarmsLanding) {
+                    // Handle dimension choices on alarms landing page
+                    parsedDimensionChoices = JSON.parse(attrs.dimensionChoices);
+                    if (stdDimensionNamespaces.indexOf(alarm.namespace) === -1) {
+                        // Alarms with custom metric/namespace
+                        parsedDimensionChoices = JSON.parse(attrs.dimensionChoices);
```
Comment (dkavanagh) [inline eucaconsole/static/js/widgets/create-alarm-modal/create-alarm.js]: seems redundant.

File: eucaconsole/static/js/widgets/create-alarm-modal/create-alarm.js [hunk 2]
```diff
@@ -146,13 +154,43 @@ angular.module('CreateAlarmModal', [
...
+                        stdDimensionNamespaces.forEach(function (namespace) {
+                            Array.prototype.push.apply(allDimensionChoices, parsedDimensionChoices[namespace]);
```
Comment (dkavanagh) [inline eucaconsole/static/js/widgets/create-alarm-modal/create-alarm.js]: better written as allDimensionChoices.push(parsedDimensionChoices[namespace]);

Comment (kamalgill) [inline eucaconsole/static/js/widgets/create-alarm-modal/create-alarm.js]: @dkavanagh: Not quite... we're joining two arrays here, and your suggestion would set the second array as the last item in the first array.

File: eucaconsole/static/js/widgets/create-alarm-modal/create-alarm.js (hunk 1 above)
Comment (kamalgill) [inline eucaconsole/static/js/widgets/create-alarm-modal/create-alarm.js]: @dkavanagh: Good catch... duplicate code removed in https://github.com/eucalyptus/eucaconsole/commit/3ded29ccf9bba

File: eucaconsole/static/js/widgets/create-alarm-modal/create-alarm.js [hunk 3]
```diff
@@ -146,13 +154,43 @@ angular.module('CreateAlarmModal', [
             };
 
             this.initializeForCopy = function (alarm, attrs) {
+                var parsedDimensionChoices = null;
+                var selectedChoices = [];
+                var allDimensionChoices = [];
+                var stdDimensionNamespaces = ['AWS/EC2', 'AWS/ELB', 'AWS/EBS'];
```
Comment (dkavanagh) [inline eucaconsole/static/js/widgets/create-alarm-modal/create-alarm.js]: AWS/AutoScaling ?

File: eucaconsole/views/alarms.py [hunk 4]
```diff
@@ -256,6 +259,13 @@ def __init__(self, request):
 
     @view_config(route_name='cloudwatch_alarms', renderer=TEMPLATE, request_method='GET')
     def alarms_landing(self):
+        dimension_options = {}
+        for namespace in ['AWS/EC2', 'AWS/ELB', 'AWS/EBS']:
```
Comment (dkavanagh) [inline eucaconsole/views/alarms.py]: AWS/AutoScaling ?

**Pull request merged**